tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableVisibilityTest.java

```diff
@@ -0,0 +1,385 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.debugger.test.adapter.variables;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.ballerinalang.debugger.test.DebugAdapterBaseTestCase;
+import org.ballerinalang.debugger.test.utils.BallerinaTestDebugPoint;
+import org.ballerinalang.debugger.test.utils.DebugUtils;
+import org.ballerinalang.test.context.BallerinaTestException;
+import org.eclipse.lsp4j.debug.StoppedEventArguments;
+import org.eclipse.lsp4j.debug.Variable;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Comparator;
+
+public class VariableVisibilityTest extends DebugAdapterBaseTestCase {
+    Variable[] variables;
+    Comparator<Variable> compareByName = Comparator.comparing(Variable::getName);
+
+    @BeforeClass
+    public void setup() {
+        testProjectName = "basic-project";
+        testModuleName = "advanced";
+        testModuleFileName = "main.bal";
+        testProjectPath = testProjectBaseDir.toString() + File.separator + testProjectName;
+        testEntryFilePath = Paths.get(testProjectPath, "src", testModuleName, testModuleFileName).toString();
+    }
+
+    @Test
+    public void parentVariableVisibilityTest() throws BallerinaTestException {
+        addBreakPoint(new BallerinaTestDebugPoint(testEntryFilePath, 150));
+        initDebugSession(DebugUtils.DebuggeeExecutionKind.RUN);
+
+        Pair<BallerinaTestDebugPoint, StoppedEventArguments> debugHitInfo = waitForDebugHit(10000);
+        variables = fetchDebugHitVariables(debugHitInfo.getRight());
+        Arrays.sort(variables, compareByName);
+
+        // var variable visibility test
+        Assert.assertEquals(variables[0].getName(), "v01_varVariable");
+        Assert.assertEquals(variables[0].getValue(), "()");
+        Assert.assertEquals(variables[0].getType(), "nil");
+
+        // boolean variable visibility test
+        Assert.assertEquals(variables[1].getName(), "v02_booleanVar");
+        Assert.assertEquals(variables[1].getValue(), "true");
+        Assert.assertEquals(variables[1].getType(), "boolean");
+
+        // int variable visibility test
+        Assert.assertEquals(variables[2].getName(), "v03_intVar");
+        Assert.assertEquals(variables[2].getValue(), "20");
+        Assert.assertEquals(variables[2].getType(), "int");
```

Comment:
It would be better if we can have one util method to assert any given variable against its name, type, and value. It will reduce the repetitiveness with better readability. WDYT?

Reply:
Yes, adding a util method would reduce the repetitiveness with better readability.
Util method added.